### ISSUE TITLE
Transcrypt: Inherit from a JS prototype?

### ISSUE THREAD

**Issue by OBKF** [QQuick/Transcrypt#330]
I am prototyping with [KiwiJS](http://www.kiwijs.org) and i it works the "JavaScript" way, but since Transcrypt uses python syntax i wanted to use the "Typescript" way using classes, [see this](http://www.kiwijs.org/documentation/using-kiwi-with-typescript-vs-javascript/extending-states-in-javascript/).
So how to inherit a Transcrypt class from a JS prototype?

In my case  i want this to work:
```
# Kiwi obviously the KiwiJS library  
class GameState(Kiwi.State):
    gameOver = False

    def checkGameOver(self):
        pass

    def update(self):
      Kiwi.State.prototype.update(this)

    def postUpdate(self):
      Kiwi.State.prototype.postUpdate(this);
      this.gameOver = this.checkGameOver();
```

by the way this works fine:
```
gameState = __new__( Kiwi.State('gameState'))

def game_preload():
    Kiwi.State.prototype.preload.call(this)
    this.addImage('logo','kiwilogo.png')

def game_create():
    Kiwi.State.prototype.create.call(this)

    this.logo = __new__(Kiwi.GameObjects.StaticImage(this, this.textures['logo'], 100, 100) )
    this.addChild(this.logo)

def game_update():
    Kiwi.State.prototype.js_update.call( this )
    this.logo.rotation += 0.01


gameState.preload = game_preload
gameState.create = game_create
gameState.js_update = game_update

game.states.addState(gameState, True)

```

Also i found this question on [stackoverflow.com](https://stackoverflow.com/questions/44213824/transcrypt-inherit-from-a-js-prototype) with same question.
thanks in advance.

**Comment by JdeH** [comment]
Hi,

I've attempted an answer at:

https://stackoverflow.com/questions/44213824/transcrypt-inherit-from-a-js-prototype/44333571#44333571

Let me know if it works for you.
Kind regards
Jacques

**Comment by OBKF** [comment]
Hello,
I was busy graduating (from yesterday, i can say i am a telecommunication engineer).
i will test it and report soon.

**Comment by OBKF** [comment]
Hello,
I was busy graduating (from yesterday, i can say i am a telecommunication engineer).
i will test it and report soon.

**Comment by JdeH** [comment]
Congratulations!

**Comment by OBKF** [comment]
Thank you, i tried to understand but this is not clear for me, can you make a simple example code ?

**Comment by doconix** [comment]
I realize this is 5 months old, but I posted a demo today that does what I think you are after.  Jacques recommends using composition, and I tried that first.  But I think I have it working in the "normal" way, although Jacques may correct my method.  But here's how I extended `React.Component`, which is a pure JS class:

```
class MyComponent(object, React.Component.prototype):
    def __init__(self, props):
        object.__init__(self)
        React.Component.apply(self, [ props ])
```

By extending both Transcrypt's `object` and the JS prototype I wanted, I get the behavior of both.  Inside __init__, I had to call the JS class' constructor by using `.apply`.  The resulting class (prototype) had attributes of both supers.

The full code is here: https://github.com/doconix/pyreact/blob/master/scripts/pyreact.py
